healthScore/tests.py

```diff
@@ -354,100 +348,151 @@ def test_view_history_requests(self):
-        url = reverse("registration")
-        request_struct = {
-            "email": "user1@example.com",
-            "username": "testUser",
-            "fullname": "test user",
-            "dob": "1990-02-02",
-            "gender": "male",
-            "street_address": "123 foo bar, foo",
-            "city": "NY",
-            "state": "NY",
-            "phone_number": "1235467890",
-        }
-        request = self.factory.post(url, request_struct)
-        response = registration(request)
-        self.assertEqual(response.status_code, 500)
 
-    def test_registration_user_username_exists_error(self):
-        url = reverse("registration")
-        request_struct = {
-            "email": "test2@test2.com",
-            "username": "user1",
-            "fullname": "test user",
-            "dob": "1990-02-02",
-            "gender": "male",
-            "street_address": "123 foo bar, foo",
-            "city": "NY",
-            "state": "NY",
-            "phone_number": "1235467890",
-        }
-        request = self.factory.post(url, request_struct)
-        response = registration(request)
-        self.assertEqual(response.status_code, 500)
+class LoginViewTest(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(
+            email="test@example.com", password="testpassword"
+        )
+
+    def test_login_view(self):
+        response = self.client.get(reverse("login"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "login.html")
+
+    def test_post_request_valid_credentials(self):
+        response = self.client.post(
+            reverse("login"), {"email": "test@example.com", "password": "testpassword"}
+        )
+        self.assertRedirects(response, reverse("homepage"))
+
+    def test_post_request_invalid_credentials(self):
+        response = self.client.post(
+            reverse("login"), {"email": "test@example.com", "password": "wrongpassword"}
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(
+            "Invalid email or password. Please try again.", response.content.decode()
+        )
+
+
+class RegistrationViewTest(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user(
+            email="test@example.com",
+            password="testpassword",
+            username="testuser",
+            name="Test User",
+        )
+
+    def test_registration_view(self):
+        response = self.client.get(reverse("registration"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "registration.html")
+
+    def test_post_request_email_exist(self):
+        response = self.client.post(
+            reverse("registration"),
+            {"email": "test@example.com", "password": "testpassword"},
+        )
+        user = User.objects.get(email="test@example.com")
+        self.assertEqual(user.email, "test@example.com")
```

Comment:
Could we please remove this assert? Since we are setting the data in the test case itself, this assert would always return True